v2/pkg/engine/plan/configuration_visitor.go

```diff
@@ -797,6 +798,9 @@ func (c *configurationVisitor) addNewPlanner(ref int, typeName, fieldName, curre
 	// fetch id is an index of the current planner
 	fetchID := len(c.planners)
 
+	// the filter needs access fieldRef
+	c.fieldRef = ref
```

Comment:
Does it needs to be reset?

Reply:
I don't think so, because it's per field. But @jensneuse or @devsergiy can weigh in.

Reply:
it will be fine like this, but maybe we should add more precise comment where this param is used

alternative is to pass field ref into resolveSubscriptionFilterCondition